sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -64,8 +64,43 @@ APIs for using IotHub v2 jobs
 
 <details><summary><b>Messages</b></summary>
 Feedback messages, sending cloud to device messages (missing from current swagger), and purging cloud to device message queue
-```csharp
 
+```csharp
+public class Messages
+{
+    /// <summary>
+    /// Retrieve feedback notification for cloud to device messages.
+    /// </summary>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>TODO: Swagger defines the response as the Http response returned from the service (and not the feedback message).</returns>
+    public virtual async Task<Response> GetCloudToDeviceMessageFeedbackAsync(CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Complete a cloud to device feedback message. A completed message is deleted from the service's feedback queue.
+    /// </summary>
+    /// <param name="lockToken">The lock token obtained when the cloud to device message was received, and provided to resolve race conditions when completing a feedback message.
+    /// TODO: lockToken is from the C2D message received on device or from message feedback received on service client?</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> CompleteCloudToDeviceMessageFeedbackAsync(string lockToken, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Abandon a cloud to device feedback message. An abandoned message is deleted from the service's feedback queue.
+    /// </summary>
+    /// <param name="lockToken">The lock token obtained when the cloud to device message was received, and provided to resolve race conditions when abandoning a feedback message.
+    /// TODO: lockToken is from the C2D message received on device or from message feedback received on service client?</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The Http response.</returns>
+    public virtual async Task<Response> AbandonCloudToDeviceMessageFeedbackAsync(string lockToken, CancellationToken cancellationToken = default) { }
+
+    /// <summary>
+    /// Purge the cloud to device message queue for a device.
+    /// </summary>
+    /// <param name="id">The unique identifier of the device.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The result of the cloud to device message queue purge operation.</returns>
+    public virtual async Task<Response<PurgeMessageQueueResult>> PurgeCloudToDeviceMessageQueueAsync(string id, CancellationToken cancellationToken = default) { }
+}
```

Comment:
Is this always for device? If so, we can call it deviceId

Reply:
I am not sure if this always only for device, but yes, the `id` is the `deviceId`, so I will rename it. 
The `PurgeMessageQueueResult` type has a parameter for module Id.